### ISSUE TITLE
LaurentPolynomial to RationalFunction 

### ISSUE THREAD

**Issue by andreasvarga** [JuliaMath/Polynomials.jl#344]
As it is now

```
julia> p = LaurentPolynomial([24,10,-15,0,1],-2,:z)
LaurentPolynomial(24*z⁻² + 10*z⁻¹ - 15 + z²)

julia> RationalFunction(p,1)
(24*z⁻² + 10*z⁻¹ - 15 + z²) // (1)
```
I wonder if it would not be better provide the rational function representation in term of standard polynomials as below
```
julia> z = Polynomial(:z)
Polynomial(1.0*z)

julia> RationalFunction(z^2*p,z^2)
(24.0 + 10.0*z - 15.0*z² + 1.0*z⁴) // (1.0*z²)
```

Also, the following (silly) construction should be possible:

```
julia> p = LaurentPolynomial([24,10,-15,0,1],-2,:z)
LaurentPolynomial(24*z⁻² + 10*z⁻¹ - 15 + z²)

julia> q = ChebyshevT([1, 0, 3, 4],:z)
ChebyshevT(1⋅T_0(z) + 3⋅T_2(z) + 4⋅T_3(z))

julia> RationalFunction(q,p)
ERROR: ArgumentError: Can't convert a Laurent polynomial with m < 0
```

Playing with polynomial conversions, I got the following result, with wrong variable:
```
julia> p = LaurentPolynomial([24,10,-15,0,1],-2,:z)
LaurentPolynomial(24*z⁻² + 10*z⁻¹ - 15 + z²)
julia> Polynomial(p)
Polynomial(24 + 10*x - 15*x^2 + x^4)
```
Since this operation determines the numerator of a Laurent polynomial seen as a rational functions, would be more appropriate to 
 define `numerator` and `denominator` for Laurent polynomials too?

Sorry if I am missing something, but is it a way to convert rational representations to standard polynomial based representations?

**Comment by jverzani** [comment]
No, you aren't missing anything. Let me try to summarize:

* There is no support for `RationalFunction(p::LaurentPolynomial, q)` ...  I would think there should be a `convert` method (`convert(RationalFunction, q::LaurentPolynomial)`) and an error thrown when trying to use the constructor.

* the problem with Chebyshev is because:
```
julia> promote_type(LaurentPolynomial{Float64,:x}, ChebyshevT{Float64,:x})
Polynomial{Float64, :x}
```

It should be like

```
julia> promote_type(LaurentPolynomial{Float64,:x}, Polynomial{Float64,:x})
LaurentPolynomial{Float64, :x}
```

As we can't convert the other way in general. The `Polynomial` type is probably caught as it is a StandardBasisPolynomial, but Chebyshev is not, so it would need to have this promotion specified.

So this example *should* be something like `convert(RationalFunction, q) // p`.

* The construction `Polynomial(p)` finds the fallback for polynomial construction from an iterable object (not a polynomial, so the symbol isn't preserved, nor is the polynomial!). The intended conversion would be `convert(Polynomial, p)`. That will error as the Laurent Polynomial has m < 0, but if m >= 0 it should preserve the symbol.

**Comment by jverzani** [comment]
I just checked in a PR which gets the following tests to pass:

```
 p = LaurentPolynomial([24,10,-15,0,1],-2,:z)
    q = ChebyshevT([1, 0, 3, 4],:z)
    @test RationalFunction(p,1) == p
    @test p //q == 1/ (q//p)
    @test numerator(p) == p * variable(p)^2
    @test denominator(p) == convert(Polynomial, variable(p)^2)
```

Does that cover everything you were expecting should work? (I'm not sure I answered your question about "a way to convert rational representations to standard polynomial based representations?")

**Comment by andreasvarga** [comment]
Yes. I think this covers the basic conversions. Still I have a question.
How can I import in a simple way a Laurent polynomial or a quotient of two
polynomials as standard rational transfer function (i.e., quotient of
standard polynomials) into the DescriptorSystems ?

john verzani ***@***.***> schrieb am Mi., 26. Mai 2021, 21:34:

**Comment by jverzani** [comment]
Maybe through the RationalFunction constructor and splatting (e.g.  if `lp` is a Laurent polynomial or `p` and `q` just standard basis polynomials

```
rational_transfer_function(lp//1..., ts)
rational_transfer_function(p//q..., ts)
```

**Comment by andreasvarga** [comment]
What I had in mind is how to "normalize" expressions like lp1//lp2, which
are common to represent so-called ARMA processes. Certainly there are
several ways to achieve it with existing functions, but a conversion
function to the p//q form would be still helpful.

john verzani ***@***.***> schrieb am So., 30. Mai 2021, 22:23:

**Comment by jverzani** [comment]
I think the same idiom would work with two Laurent Polynomials, `rational_transfer_function(lp//lq..., ts)`. To cancel common factors first, you would use `lowest_terms(lp//lpq)...`. (Is that what you have in mind by "normalize", or is there some other canonical form?)

**Comment by andreasvarga** [comment]
With "normalize" I mean to represent a quotient of two arbitrary
polynomials, in any basis, including Laurent polynomials, as a ratio of two
polynomials in the standard monomial basis.

john verzani ***@***.***> schrieb am So., 30. Mai 2021, 22:41:

**Comment by jverzani** [comment]
Then that `p//q...` trick should do work for that. Let me know if it doesn't or it requires more manipulations than that.

**Comment by jverzani** [comment]
Closed by #345
